src/modules/LanItems.rb

```diff
@@ -1471,7 +1471,7 @@ def FormatOverview
         overviews[iface.name] = {}
         rich = ""
 
-        descr = iface.name
+        descr = iface.hardware.description
```

Comment:
Virtual interfaces do not have have harward info, ins't it?

Reply:
well ... depends. As I see it ```hwinfo``` provides more a kind of "driver" information provided by kernel. So, if you use ```hwinfo --network``` instead of ```hwinfo --netcard``` you'll see that (e.g.) vlan device can be found like on one of my test VM:
```
17: None 00.0: 10701 Ethernet
  [Created at net.126]
  Unique ID: +HKJ.ndpeucax6V1
  Parent ID: rBUF.GkSpXDNdaaB
  SysFS ID: /class/net/vlan0
  Hardware Class: network interface
  Model: "Ethernet network interface"
  Driver: "802.1Q VLAN Support"
  Device File: vlan0
  HW Address: 08:00:27:57:ed:fe
  Permanent HW Address: 08:00:27:57:ed:fe
  Link detected: yes
  Config Status: cfg=new, avail=yes, need=no, active=unknown
  Attached to: #5 (Ethernet controller
```

On the other hand yast2-network currently queries only for ```-netcard``` which do not show the vlan device in this case. 

I only try to separate access to ```LanItems::Items``` and wrap it to some objects here. It definitely is not perfect so far and needs some love. Moreover the piece of code you refer to is a direct "translation" of old code into new - in original code it simply put empty string into ```descr``` when corresponding hwitem field (```hwinfo["name"]``` - a textual description of device like "Intel 82546 controller") was not available for the device.